components/location/popup.jsx

```diff
@@ -0,0 +1,319 @@
+  const url_pattern = new RegExp(
+    /\b/.source + // all URLs start after a word boundary
+      "(" +
+      web_pattern +
+      "|" +
+      phone_pattern.source +
+      "|" +
+      email_pattern.source +
+      ")",
+    "gi" // find all occurrences, case-insensitive
+  );
+
+  return item.match(url_pattern);
+}
+
+function parseLineBreaks(value) {
+  return value.replace(/\n/g, "<br />");
+}
+
+const PopupSection = ({ content, id }) => {
+  const { getTranslation } = useContext(LanguageContext);
+
+  return (
+    <div className="p row">
+      <p
+        // data-translation-id="${_.snakeCase(id)}"
+        className="key"
+      >
+        {getTranslation(id)}
+      </p>
+      {/** Kanad Note: I had to change this from a p tag to a div tag
+       * since it was causing issues when doing a ReactDOMServer render */}
+      <div className="value">{content}</div>
+    </div>
+  );
+};
+
+const NeedsMoney = ({ location }) => {
+  const { getTranslation } = useContext(LanguageContext);
+
+  let link = "";
+  if (location.seekingMoneyURL && location.seekingMoneyURL !== "") {
+    link = (
+      <a
+        data-translation-id="seeking_money_link"
+        href="${location.seekingMoneyURL}"
+        target="_blank"
+        // onclick="captureOutboundLink('${location.seekingMoneyURL}', 'donation')"
+      >
+        {getTranslation("seeking_money_link")}
+      </a>
+    );
+  }
+  return (
+    <span className="seekingMoney seeking-money card-badge">
+      <span data-translation-id="seeking_money">
+        {getTranslation("seeking_money")}
+      </span>{" "}
+      {link}
+    </span>
+  );
+};
+
+const OpenHours = ({ closingHours, id, openingHours }) => {
+  if (openingHours === "never" || openingHours === "not today") {
+    return <PopupSection content={openingHours} id={id} />;
+  } else {
+    const openHours = closingHours.map((closeHour, i) => {
+      <p>
+        <span>{openingHours[i]}</span> to <span>{closingHours[i]}</span>
+      </p>;
+    });
+    return <PopupSection content={openHours} id={id} />;
+  }
+};
+
+const NoIdNeeded = ({ location }) => {
+  const { getTranslation } = useContext(LanguageContext);
+
+  if (location.someInfoRequired === true) {
+    return (
+      <span
+        className="noIdNeeded card-badge"
+        // data-translation-id="some_info_required"
+      >
+        {getTranslation("some_info_required")}
+      </span>
+    );
+  } else {
+    return (
+      <span
+        className="noIdNeeded card-badge"
+        // data-translation-id="no_id_needed"
+      >
+        {getTranslation("no_id_needed")}
+      </span>
+    );
+  }
+};
+
+const WarmingSite = () => {
+  const { getTranslation } = useContext(LanguageContext);
+  return (
+    <span
+      // Kanad note: this translation ID is wrong!
+      // data-translation-id="warming_site"
+      className="warmingSite card-badge"
+    >
+      {getTranslation("warm_up_here")}
+    </span>
+  );
+};
+
+// const Address = ({ address }) => {
```

Comment:
??

Reply:
Welp this fell through the cracks—will remove! I don't remember why but I think I discovered that this didn't need to be its own component, hence it's implemented here: https://github.com/Twin-Cities-Mutual-Aid/twin-cities-aid-distribution-locations/blob/8cb738911293f86aa7de13834a6da55e0551b033/components/location/popup.jsx#L233-L244

Reply:
Removed! https://github.com/Twin-Cities-Mutual-Aid/twin-cities-aid-distribution-locations/pull/302/commits/67aa95ff7a62ea897e52e79d7ee7dfe17b7081c3